--- slurm.spec	2013-07-30 16:12:29.000000000 -0500
+++ native-slurm.spec	2013-07-30 18:08:30.000000000 -0500
@@ -37,6 +37,15 @@
 #
 %define slurm_with() %{expand:%%{?slurm_with_%{1}:1}%%{!?slurm_with_%{1}:0}}
 
+# Define some defaults for rpmbuild
+%define _prefix /usr/local
+%define _sysconfdir %{_prefix}/etc/slurm
+%define _mandir %{_prefix}/share/man
+%define _infodir %{_prefix}/share/info
+
+# This needs to be defined for incremental builds
+%define intranamespace_name slurm
+
 #  Options that are off by default (enable with --with <opt>)
 %slurm_without_opt auth_none
 %slurm_without_opt authd
@@ -92,16 +101,16 @@
 %slurm_with_opt sgijob
 %endif
 
-Name:    see META file
-Version: see META file
-Release: see META file
+Name:    slurm 
+Version: 13.12.0 
+Release: 0pre1 
 
 Summary: Simple Linux Utility for Resource Management
 
 License: GPL
 Group: System Environment/Base
-Source: %{name}-%{version}-%{release}.tgz
-BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}
+Source: %{name}-%{version}.tar.bz2
+BuildRoot: %{_tmppath}/%{name}-%{version}
 URL: http://slurm.schedmd.com/
 
 Requires: slurm-plugins
@@ -156,6 +165,16 @@
 Requires: cray-MySQL-devel-enterprise
 %endif
 
+# Add requirements for build
+%if %{slurm_with cray}
+BuildRequires: cray-MySQL-devel-enterprise
+BuildRequires: cray-libalpslli-devel
+BuildRequires: cray-libalpscomm_cn-devel
+BuildRequires: cray-libalpscomm_sn-devel
+BuildRequires: cray-libjob-devel
+BuildRequires: -post-build-checks
+%endif
+
 %ifnos aix5.3
 # FIXME: AIX can't seem to find this even though this is in existance there.
 # We should probably figure out a better way of doing this, but for now we
@@ -235,11 +254,17 @@
 %package devel
 Summary: Development package for SLURM
 Group: Development/System
-Requires: slurm
+Requires: slurm-libs
 %description devel
 Development package for SLURM.  This package includes the header files
 and static libraries for the SLURM API
 
+%package libs
+Summary: SLURM libraries
+Group: Development/System
+%description libs
+SLURM libraries, for building and running programs using the SLURM API.
+
 %if %{slurm_with auth_none}
 %package auth-none
 Summary: SLURM auth NULL implementation (no authentication)
@@ -408,9 +433,11 @@
 #############################################################################
 
 %prep
-%setup -n %{name}-%{version}-%{release}
+%incremental_setup -n %{name}-%{version}
 
 %build
+# Skip configure if possible
+if [ ! -f "config.status" -o "%{reconfigure}" = "1" ]; then
 %configure \
 	%{?slurm_with_debug:--enable-debug} \
 	%{?slurm_with_partial_attach:--enable-partial-attach} \
@@ -430,6 +457,7 @@
 	%{?slurm_with_salloc_background:--enable-salloc-background} \
 	%{!?slurm_with_readline:--without-readline} \
 	%{?with_cflags}
+fi
 
 make %{?_smp_mflags}
 
@@ -451,9 +479,7 @@
 %endif
 
 %if %{slurm_with cray} || %{slurm_with cray_alps}
-   if [ -d /opt/modulefiles ]; then
-      install -D -m644 contribs/cray/opt_modulefiles_slurm $RPM_BUILD_ROOT/opt/modulefiles/slurm/opt_modulefiles_slurm
-   fi
+   install -D -m644 contribs/cray/opt_modulefiles_slurm $RPM_BUILD_ROOT/opt/modulefiles/slurm
 %else
    rm -f contribs/cray/opt_modulefiles_slurm
 %endif
@@ -559,9 +585,6 @@
 test -f $RPM_BUILD_ROOT/usr/sbin/rcslurm			&&
   echo /usr/sbin/rcslurm				>> $LIST
 
-test -f $RPM_BUILD_ROOT/opt/modulefiles/slurm/opt_modulefiles_slurm &&
-  echo /opt/modulefiles/slurm/opt_modulefiles_slurm	>> $LIST
-
 # Make ld.so.conf.d file
 mkdir -p $RPM_BUILD_ROOT/etc/ld.so.conf.d
 echo '%{_libdir}
@@ -707,7 +730,6 @@
 %ifos aix5.3
 %{_sbindir}/srun
 %endif
-%{_libdir}/*.so*
 %{_libdir}/slurm/src/*
 %{_mandir}/man1/*
 %{_mandir}/man5/acct_gather.*
@@ -724,10 +746,8 @@
 %{_mandir}/man8/spank*
 %dir %{_sysconfdir}
 %dir %{_libdir}/slurm/src
-%dir /etc/ld.so.conf.d
-/etc/ld.so.conf.d/slurm.conf
 %if %{slurm_with cray} || %{slurm_with cray_alps}
-%dir /opt/modulefiles/slurm
+/opt/modulefiles/slurm
 %endif
 %config %{_sysconfdir}/slurm.conf.example
 %config %{_sysconfdir}/cgroup.conf.example
@@ -759,6 +779,14 @@
 #%{_mandir}/man3/slurmdb_*
 #############################################################################
 
+%files libs
+%defattr(-,root,root)
+%dir %attr(0755,root,root)
+%{_libdir}/*.so*
+%dir /etc/ld.so.conf.d
+/etc/ld.so.conf.d/slurm.conf
+#############################################################################
+
 %if %{slurm_with auth_none}
 %files auth-none
 %defattr(-,root,root)
@@ -974,7 +1002,7 @@
 #    fi
 #fi
 
-%post
+%post libs
 if [ -x /sbin/ldconfig ]; then
     /sbin/ldconfig %{_libdir}
     if [ $1 = 1 ]; then
@@ -1009,7 +1037,7 @@
     fi
 fi
 
-%postun
+%postun libs
 if [ "$1" = 0 ]; then
     if [ -x /sbin/ldconfig ]; then
 	/sbin/ldconfig %{_libdir}
